code/zk-game/contracts/deploy/utils.ts

```diff
@@ -0,0 +1,177 @@
+ * @param {string} data.contract The contract's path and name. E.g., "contracts/Greeter.sol:Greeter"
+ */
+export const verifyContract = async (data: {
+  address: string;
+  contract: string;
+  constructorArguments: string;
+  bytecode: string;
+}) => {
+  const verificationRequestId: number = await hre.run('verify:verify', {
+    ...data,
+    noCompile: true,
+  });
+  return verificationRequestId;
+};
+
+type DeployContractOptions = {
+  /**
+   * If true, the deployment process will not print any logs
+   */
+  silent?: boolean;
+  /**
+   * If true, the contract will not be verified on Block Explorer
+   */
+  noVerify?: boolean;
+  /**
+   * If specified, the contract will be deployed using this wallet
+   */
+  wallet?: Wallet;
+};
+export const deployContract = async (
+  contractArtifactName: string,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  constructorArguments?: any[],
+  options?: DeployContractOptions
+) => {
+  const log = (message: string) => {
+    if (!options?.silent) console.log(message);
+  };
+
+  log(`\nStarting deployment process of "${contractArtifactName}"...`);
+
+  const wallet = options?.wallet ?? getWallet();
+  const deployer = new Deployer(hre, wallet);
+  const artifact = await deployer.loadArtifact(contractArtifactName).catch((error) => {
+    if (error?.message?.includes(`Artifact for contract "${contractArtifactName}" not found.`)) {
+      console.error(error.message);
+      throw `⛔️ Please make sure you have compiled your contracts or specified the correct contract name!`;
+    } else {
+      throw error;
+    }
+  });
+
+  // Estimate contract deployment fee
+  const deploymentFee = await deployer.estimateDeployFee(artifact, constructorArguments || []);
+  log(`Estimated deployment cost: ${ethers.formatEther(deploymentFee)} ETH`);
+
+  // Check if the wallet has enough balance
+  await verifyEnoughBalance(wallet, deploymentFee);
+
+  // Deploy the contract to ZKsync
+  const contract = await deployer.deploy(artifact, constructorArguments);
+  const address = await contract.getAddress();
+  const constructorArgs = contract.interface.encodeDeploy(constructorArguments);
+  const fullContractSource = `${artifact.sourceName}:${artifact.contractName}`;
+
+  // Display contract deployment info
+  log(`\n"${artifact.contractName}" was successfully deployed:`);
+  log(` - Contract address: ${address}`);
+  log(` - Contract source: ${fullContractSource}`);
+  log(` - Encoded constructor arguments: ${constructorArgs}\n`);
+
+  if (!options?.noVerify && hre.network.config.verifyURL) {
+    log(`Requesting contract verification...`);
+    await verifyContract({
+      address,
+      contract: fullContractSource,
+      constructorArguments: constructorArgs,
+      bytecode: artifact.bytecode,
+    });
+  }
+
+  return contract;
+};
+
+/**
+ * Rich wallets can be used for testing purposes.
+ * Available on ZKsync In-memory node and Dockerized node.
+ */
+export const LOCAL_RICH_WALLETS = [
```

Comment:
no need to hardhcode this if we use the `getSigners()` method. Example here: https://github.com/matter-labs/paymaster-examples/blob/70e44c28ea023298d930e4ab7a90a41efcb5a6ca/contracts/deploy/applicationPaymaster.ts#L25